src/main/java/de/dennisguse/opentracks/publicapi/CreateMarkerActivity.java

```diff
@@ -0,0 +1,82 @@
+package de.dennisguse.opentracks.publicapi;
+
+import android.content.Intent;
+import android.location.Location;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.time.Instant;
+
+import de.dennisguse.opentracks.AbstractActivity;
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.data.models.Marker;
+import de.dennisguse.opentracks.data.models.Track;
+import de.dennisguse.opentracks.data.models.TrackPoint;
+import de.dennisguse.opentracks.services.TrackRecordingService;
+import de.dennisguse.opentracks.services.TrackRecordingServiceConnection;
+import de.dennisguse.opentracks.ui.markers.MarkerEditActivity;
+import de.dennisguse.opentracks.util.IntentUtils;
+
+/**
+ * Public API to creates a Marker for a given track with a given location
+ */
+public class CreateMarkerActivity extends AppCompatActivity {
+
+    public static final String EXTRA_TRACK_ID = "track_id";
+    public static final String EXTRA_LOCATION = "location";
+
+    private static final String TAG = CreateMarkerActivity.class.getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // TODO: do we want to check if public API is enabled?
+
+        Track.Id trackId;
+        if (getIntent().hasExtra(EXTRA_TRACK_ID)) {
+            trackId = new Track.Id(getIntent().getLongExtra(EXTRA_TRACK_ID, 0L));
+        } else {
+            trackId = null;
+        }
+        TrackPoint trackPoint;
+        if (getIntent().hasExtra(EXTRA_LOCATION)) {
+            Location location = getIntent().getParcelableExtra(EXTRA_LOCATION);
+            trackPoint = new TrackPoint(location, Instant.now());
+        } else {
+            trackPoint = null;
+        }
+
+        TrackRecordingServiceConnection.execute(this, (service, self) -> {
+            Marker.Id marker = createNewMarker(trackId, trackPoint, service);
```

Comment:
Why not pass in the location?
Or rather the value object that we now have in OpenTracks. 
The trackpoint shouldnt be required.

Reply:
Unfortunately Marker need  `accuracy`, `altitude` and  `bearing` from the TrackPoint. Or should we remove them, too?

Reply:
https://github.com/OpenTracksApp/OpenTracks/blob/main/src%2Fmain%2Fjava%2Fde%2Fdennisguse%2Fopentracks%2Fdata%2Fmodels%2FPosition.java

Reply:
Thanks, was searching for Location